packages/world-modules/src/utils/AccessControlLib.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { ResourceId, WorldResourceIdInstance } from "@latticexyz/world/src/WorldResourceId.sol";
+import { IWorldErrors } from "@latticexyz/world/src/IWorldErrors.sol";
+import { ResourceAccess } from "@latticexyz/world/src/codegen/tables/ResourceAccess.sol";
+import { NamespaceOwner } from "@latticexyz/world/src/codegen/tables/NamespaceOwner.sol";
+
+/**
+ * @title AccessControlLib
+ * @dev Provides access control functions for checking permissions and ownership within a namespace.
+ * This library is functionally equivalent with the AccessControl library from world,
+ * but uses StoreSwitch instead of always reading from own storage.
+ */
+library AccessControlLib {
```

Comment:
Though I understand why it's in `world-modules`, would it maybe be better to have a single `AccessControl` lib with `hasAccess`/`_hasAccess` etc, to have all the logic in 1 place and let it mirror the table methods?

Reply:
agree! unfortunately can't touch the code in `world` while it's being audited but should note this down for later

Reply:
https://github.com/latticexyz/mud/issues/1853